cmd/dep/godep_importer.go

```diff
@@ -0,0 +1,169 @@
+}
+
+func newGodepImporter(logger *log.Logger, verbose bool, sm gps.SourceManager) *godepImporter {
+	return &godepImporter{
+		logger:  logger,
+		verbose: verbose,
+		sm:      sm,
+	}
+}
+
+type godepJSON struct {
+	Name    string         `json:"ImportPath"`
+	Imports []godepPackage `json:"Deps"`
+}
+
+type godepPackage struct {
+	ImportPath string `json:"ImportPath"`
+	Rev        string `json:"Rev"`
+	Comment    string `json:"Comment"`
+}
+
+func (g *godepImporter) Name() string {
+	return "godep"
+}
+
+func (g *godepImporter) HasDepMetadata(dir string) bool {
+	y := filepath.Join(dir, "Godeps", godepJSONName)
+	if _, err := os.Stat(y); err != nil {
+		return false
+	}
+
+	return true
+}
+
+func (g *godepImporter) Import(dir string, pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	err := g.load(dir)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return g.convert(pr)
+}
+
+func (g *godepImporter) load(projectDir string) error {
+	g.logger.Println("Detected godep configuration files...")
+	j := filepath.Join(projectDir, "Godeps", godepJSONName)
+	if g.verbose {
+		g.logger.Printf("  Loading %s", j)
+	}
+	jb, err := ioutil.ReadFile(j)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to read %s", j)
+	}
+	err = json.Unmarshal(jb, &g.json)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to parse %s", j)
+	}
+
+	return nil
+}
+
+func (g *godepImporter) convert(pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	g.logger.Println("Converting from Godeps.json ...")
+
+	manifest := &dep.Manifest{
+		Constraints: make(gps.ProjectConstraints),
+	}
+	lock := &dep.Lock{}
+
+	for _, pkg := range g.json.Imports {
+		// ImportPath must not be empty
+		if pkg.ImportPath == "" {
+			err := errors.New("Invalid godep configuration, ImportPath is required")
+			return nil, nil, err
+		}
+
+		// Rev must not be empty
+		if pkg.Rev == "" {
+			err := errors.New("Invalid godep configuration, Rev is required")
+			return nil, nil, err
+		}
+
+		if pkg.Comment == "" {
+			// When there's no comment, try to get corresponding version for the Rev
+			// and fill Comment.
+			pi := gps.ProjectIdentifier{
+				ProjectRoot: gps.ProjectRoot(pkg.ImportPath),
+			}
+			revision := gps.Revision(pkg.Rev)
+			version, err := lookupVersionForRevision(revision, pi, g.sm)
+			if err != nil {
+				warn := errors.Wrapf(err, "Unable to lookup the version represented by %s in %s. Falling back to locking the revision only.", pkg.Rev, pi.ProjectRoot)
+				g.logger.Printf(warn.Error())
+				version = revision
+			}
+
+			pp := getProjectPropertiesFromVersion(version)
+			if pp.Constraint != nil {
+				pkg.Comment = pp.Constraint.String()
+			}
+		}
+
+		if pkg.Comment != "" {
+			// If there's a comment, use it to create project constraint
+			pc, err := g.buildProjectConstraint(pkg)
+			if err != nil {
+				return nil, nil, err
+			}
+			manifest.Constraints[pc.Ident.ProjectRoot] = gps.ProjectProperties{Source: pc.Ident.Source, Constraint: pc.Constraint}
```

Comment:
Does godep allow specifying an alternative source? I didn't see anything handling that in `buildProjectConstraint`. If not, we can skip setting the `Source` field on this line since it will always be `""`.